cardano-node/src/Cardano/Node/ToObjectOrphans.hs

```diff
@@ -0,0 +1,181 @@
+            mkObject [ "kind" .= String "TraceAddBlockEvent.SwitchedToChain"
+                     , "newtip" .= showTip verb (AF.headPoint c) ]
+        ChainDB.AddBlockValidation ev' -> case ev' of
+            ChainDB.InvalidBlock err pt ->
+                mkObject [ "kind" .= String "TraceAddBlockEvent.AddBlockValidation.InvalidBlock"
+                         , "block" .= toObject verb pt
+                         , "error" .= show err ]
+            _ -> emptyObject
+        _ -> emptyObject
+
+    toObject MinimalVerbosity (ChainDB.TraceLedgerReplayEvent _ev) = emptyObject -- no output
+    toObject verb (ChainDB.TraceLedgerReplayEvent ev) = case ev of
+        LedgerDB.ReplayFromGenesis _replayTo ->
+            mkObject [ "kind" .= String "TraceLedgerReplayEvent.ReplayFromGenesis" ]
+        LedgerDB.ReplayFromSnapshot snap tip' _replayTo ->
+            mkObject [ "kind" .= String "TraceLedgerReplayEvent.ReplayFromSnapshot"
+                     , "snapshot" .= toObject verb snap
+                     , "tip" .= show tip' ]
+        _ -> emptyObject
+
+    toObject MinimalVerbosity (ChainDB.TraceLedgerEvent _ev) = emptyObject -- no output
+    toObject verb (ChainDB.TraceLedgerEvent ev) = case ev of
+        LedgerDB.TookSnapshot snap pt ->
+            mkObject [ "kind" .= String "TraceLedgerEvent.TookSnapshot"
+                     , "snapshot" .= toObject verb snap
+                     , "tip" .= show pt ]
+        LedgerDB.DeletedSnapshot snap ->
+            mkObject [ "kind" .= String "TraceLedgerEvent.DeletedSnapshot"
+                     , "snapshot" .= toObject verb snap ]
+        LedgerDB.InvalidSnapshot snap failure ->
+            mkObject [ "kind" .= String "TraceLedgerEvent.InvalidSnapshot"
+                     , "snapshot" .= toObject verb snap
+                     , "failure" .= show failure ]
+
+    toObject verb (ChainDB.TraceCopyToImmDBEvent ev) = case ev of
+        ChainDB.CopiedBlockToImmDB pt ->
+            mkObject [ "kind" .= String "TraceCopyToImmDBEvent.CopiedBlockToImmDB"
+                     , "slot" .= toObject verb pt ]
+        _ -> emptyObject
+
+    toObject verb (ChainDB.TraceGCEvent ev) = case ev of
+        ChainDB.PerformedGC slot ->
+            mkObject [ "kind" .= String "TraceGCEvent.PerformedGC"
+                     , "slot" .= toObject verb slot ]
+        _ -> emptyObject
+
+    toObject verb (ChainDB.TraceOpenEvent ev) = case ev of
+        ChainDB.OpenedDB immTip tip' ->
+            mkObject [ "kind" .= String "TraceOpenEvent.OpenedDB"
+                     , "immtip" .= toObject verb immTip
+                     , "tip" .= toObject verb tip' ]
+        _ -> emptyObject
+
+    toObject _ _ = emptyObject -- no output
+
+instance ToObject LedgerDB.DiskSnapshot where
+    toObject MinimalVerbosity snap = toObject NormalVerbosity snap
+    toObject NormalVerbosity _ = mkObject [ "kind" .= String "snapshot" ]
+    toObject MaximalVerbosity snap =
```

Comment:
Is this the only place where `MaximalVerbosity` does something different than `NormalVerbosity`?

I was expecting that `MaximalVerbosity` would just trace *all* trace events. This would make debugging much easier: just enable enable maximal verbosity with a flag, instead of modifying the code (to use, e.g., `showTracing debugTracer`), and recompiling. Or is this not the purpose of `MaximalVerbosity`?

Reply:
yes, probably. This needs to be discussed, then can be changed.